dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -245,4 +245,36 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
             LOG_IF_FAILED(::CoRevokeClassObject(static_cast<DWORD>(token.Cookie())));
         }
     }
+
+    bool PushNotificationManager::IsBIAvailable()
+    {
+        try
+        {
+            BackgroundTaskBuilder builder = BackgroundTaskBuilder();
+            auto builder5 = builder.as<winrt::IBackgroundTaskBuilder5>();
+        }
+        catch (...)
+        {
+            return false;
```

Comment:
Silently swallowing errors is rarely wise. At least a LOG_ERROR_MSG() here so there's a trail if something goes wrong.

Imagine you have to support this code, a year from now. In the wild.
Imagine a production system is affected. There is no debugger or or invasive process.
Imagine this fails. Users don't know, other than the end behavior just doesn't happen.
What do you do?

Logs are your best friend. Learn to love them :-)

Reply:
This is a scenario that we support where packaged apps don't have access to IBackgroundTaskBuilder5 for builds RS5 or below. Instead, they will call RegisterActivator with the ProtocolFlag to be launched by our LRP.

Because this is a scenario we support, should we LOG_ERROR_MSG() still?

Reply:
Is there a way to perform a QI that doesn't throw exceptions? I think that would be ideal here. 

I do agree with the notion that it is strange pattern to rely on exceptions for cases like this.

Reply:
Ok did some digging. Instead of .as, you can use try_as. It will return nullptr if the QI fails and you can check against that instead of relying on an exception. Cleaner code.